src/js/components/modals/InstallPackageModal.js

```diff
@@ -89,19 +92,29 @@ class InstallPackageModal
         currentTab: "defaultInstall",
         truncatedPreInstallNotes: true
       });
+
+      return;
     }
-    // If Opening
-    if (!props.open && nextProps.open) {
-      // If default - auto-install
-      if (!nextProps.advancedConfig) {
-        this.handleInstallPackage();
-      } else {
-        // Show advanced install
-        this.setState({
-          currentTab: "advancedInstall"
-        });
-      }
+
+    if (nextProps.isBetaPackage) {
+      this.setState({
+        betaTermsAccepted: false,
+        betaOptInProperties: BetaOptInUtil.getProperty(
+          nextProps.cosmosPackage.getConfig()
+        ),
+        currentTab: "betaTerms"
+      });
+
+      return;
     }
+
+    if (nextProps.advancedConfig) {
+      this.setState({ currentTab: "advancedInstall" });
+
+      return;
+    }
+
+    this.handleInstallPackage();
```

Comment:
It feels a little odd to call this method at the end of this lifecycle method. Could you write a comment about why this is needed? This method also ends up calling `#forceUpdate` which we should avoid calling throughout the lifecycle.